iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/orgalternativenames/AffMatchOrganizationAltNameFiller.java

```diff
@@ -0,0 +1,85 @@
+package eu.dnetlib.iis.wf.affmatching.orgalternativenames;
+
+import static java.util.stream.Collectors.toList;
+
+import java.io.Serializable;
+import java.util.List;
+
+import com.google.common.collect.Lists;
+
+import eu.dnetlib.iis.wf.affmatching.model.AffMatchOrganization;
+
+/**
+ * Filler of {@link AffMatchOrganization#getAlternativeNames()} list
+ * based on a dictionary of alternative organization names.
+ * 
+ * @author madryk
+ */
+public class AffMatchOrganizationAltNameFiller implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+    
+    
+    private List<List<String>> alternativeNamesDictionary = Lists.newArrayList();
+    
+
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Fills {@link AffMatchOrganization#getAlternativeNames()} list with
+     * alternative names found in dictionary.<br/>
+     * Method is looking for alternative names in dictionary
+     * based on equality of {@link AffMatchOrganization#getName()}
+     * or any of the {@link AffMatchOrganization#getAlternativeNames()} that
+     * was already present in passed organization.
+     * 
+     * Notice that method modifies {@link AffMatchOrganization#getAlternativeNames()}
+     * of the passed organization.<br/>
+     */
+    public AffMatchOrganization fillAlternativeNames(AffMatchOrganization organization) {
+        
+        List<String> alternativeNames = fetchAlternativeNames(organization.getNames());
+        
+        alternativeNames.stream().forEach(altName -> organization.addAlternativeName(altName));
+        
+        return organization;
+    }
+    
+    
+    //------------------------ PRIVATE --------------------------
+    
+    private List<String> fetchAlternativeNames(List<String> originalNames) {
+        
+        for (List<String> alternativeNames : alternativeNamesDictionary) {
+            
+            if (containsSameOrganizationName(originalNames, alternativeNames)) {
+                return filterAlreadyPresentAlternativeNames(alternativeNames, originalNames);
```

Comment:
Not sure if I understand it right.

There is no need to always go through all alternativeNames. We assume that single organization alternative names are defined in one line and only in single file. The same organization can't be added to dictionary twice with different alternative names. So first match is good enough. Otherwise some merging mechanism should be implemented when building dictionary.

Yes, we always stop after finding first match (in any of the alternative names). We're looking in all of alternative names so it doesn't matter if its English or native.

If affiliation is in English and organization is in native language. Then to organization alternative names English version will be added. Next we will have to update voters, so they will use alternativeName field. And match will be found.